### DISCUSSION TITLE
Microsoft.Extensions.Http.Resilience - What is the Max Value Possible For QueueLimit Of ConcurrencyLimiterOptions?

### DISCUSSION THREAD

**Discussion by Sn3akyP3t3** [dotnet/extensions#5566] [Q&A]
I'm unclear what the max value allowed for QueueLimit Of ConcurrencyLimiterOptions is allowed to be.

I see documentation states that [the property is an int32](https://learn.microsoft.com/en-us/dotnet/api/system.threading.ratelimiting.concurrencylimiteroptions.queuelimit?view=aspnetcore-8.0#system-threading-ratelimiting-concurrencylimiteroptions-queuelimit) so does that hint that it can be Int32.MaxValue?

**Comment by iliar-turdushev** [comment]
Hello @Sn3akyP3t3,

We do not introduce any limitations on the `QueueLimit` property in the `Microsoft.Extensions.Http.Resilience`. Neither does Polly v8 which we use under the hood to implement rate limiting for HttpClient. I also checked dotnet/runtime repo where [ConcurrencyLimiterOptions](https://github.com/dotnet/runtime/blob/a0fdddab98ad95186d84d4667df4db8a4e651990/src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/ConcurrencyLimiterOptions.cs) is originated from, and they don't do have any limitations on `QueueLimit` as well, except that it must be `>=0`. So, I believe your right and it can be Int32.MaxValue.
